client/manager/manager.go

```
@@ -0,0 +1,154 @@
+	}
+}
+
+func (m *ClientManager) New(meta oidc.ClientMetadata) (*oidc.ClientCredentials, error) {
+	tx, err := m.begin()
+	if err != nil {
+		return nil, err
+	}
+	defer rollback(tx)
+
+	// Generate Client ID
+	clientID, err := m.clientIDGenerator(meta.RedirectURIs[0].Host)
+
+	// Generate Secret
+	secret, err := m.secretGenerator()
+	if err != nil {
+		return nil, err
+	}
+	clientSecret := base64.URLEncoding.EncodeToString(secret)
+	cli := client.Client{
+		Credentials: oidc.ClientCredentials{
+			ID:     clientID,
+			Secret: clientSecret,
+		},
+		Metadata: meta,
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Save Client
+	cc, err := m.clientRepo.New(tx, cli)
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+	return cc, nil
+}
+
+func (m *ClientManager) Get(id string) (client.Client, error) {
+	return m.clientRepo.Get(nil, id)
+}
+
+func (m *ClientManager) All() ([]client.Client, error) {
+	return m.clientRepo.All(nil)
+}
+
+func (m *ClientManager) Metadata(clientID string) (*oidc.ClientMetadata, error) {
+	c, err := m.clientRepo.Get(nil, clientID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &c.Metadata, nil
+}
+
+func (m *ClientManager) IsDexAdmin(clientID string) (bool, error) {
+	c, err := m.clientRepo.Get(nil, clientID)
+	if err != nil {
+		return false, err
+	}
+
+	return c.Admin, nil
+}
+
+func (m *ClientManager) SetDexAdmin(clientID string, isAdmin bool) error {
+	tx, err := m.begin()
+	if err != nil {
+		return err
+	}
+	defer rollback(tx)
+
+	c, err := m.clientRepo.Get(tx, clientID)
+	if err != nil {
+		return err
+	}
+
+	c.Admin = isAdmin
+	err = m.clientRepo.Update(tx, c)
+	if err != nil {
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *ClientManager) Authenticate(creds oidc.ClientCredentials) (bool, error) {
+	clientSecret, err := m.clientRepo.GetSecret(nil, creds.ID)
+	if err != nil || clientSecret == nil {
```

Why are we swallowing this error rather than returning one?